test/parallel/test-http-upgrade-agent.js

```diff
@@ -46,9 +46,14 @@ srv.listen(common.PORT, '127.0.0.1', function() {
   req.end();
 
   req.on('upgrade', function(res, socket, upgradeHead) {
-    // XXX: This test isn't fantastic, as it assumes that the entire response
-    //      from the server will arrive in a single data callback
-    assert.equal(upgradeHead, 'nurtzo');
+    var recvData = upgradeHead;
+    socket.on('data', function(d) {
+      recvData += d;
+    });
+
+    socket.on('close', function() {
```

Comment:
Nit: As with the other PR that's similar to this, this callback might benefit from being wrapped in `common.mustCall()` to guarantee that it fires.